### PR TITLE
Added CinemaHall-, Order-, User-, and  ShoppingCartControllers

### PR DESCRIPTION
**Pull request by julia-shvabovich**

### CODE REVIEW

File: src/main/java/cinema/controller/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    @PostMapping("/complete")
+    public void completeOrder(@RequestBody String email) {
+        ShoppingCart shoppingCart = shoppingCartService
+                .getByUser(userService.findByEmail(email).get());
+        orderService.completeOrder(shoppingCart.getTickets(), shoppingCart.getUser());
+    }
+
+    @GetMapping
+    public List<OrderResponseDto> getOrdersHistory(User user) {
```
Comment (fyarkiy) [inline src/main/java/cinema/controller/OrderController.java]: I would rather think you will have history based on userId as @requestParam

File: src/main/java/cinema/controller/ShoppingCartController.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                                  ShoppingCartDtoMapper shoppingCartDtoMapper) {
+        this.shoppingCartService = shoppingCartService;
+        this.userService = userService;
+        this.movieSessionDtoMapper = movieSessionDtoMapper;
+        this.shoppingCartDtoMapper = shoppingCartDtoMapper;
+    }
+
+    @PostMapping("/movie-sessions")
+    public void addMovieSession(@RequestParam String userEmail,
+                                @RequestBody MovieSessionRequestDto movieSession) {
```
Comment (fyarkiy) [inline src/main/java/cinema/controller/ShoppingCartController.java:37]: I wasn't able to get it works with moviesSession - are you getting MovieSessionId in your RequestDto?

File: src/main/java/cinema/model/dto/ShoppingCartResponseDto.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
+package cinema.model.dto;
+
+import cinema.model.Ticket;
+import java.util.List;
+
+public class ShoppingCartResponseDto {
+    private Long id;
+    private List<Ticket> tickets;
```
Comment (fyarkiy) [inline src/main/java/cinema/model/dto/ShoppingCartResponseDto.java]: it's a bit tricky to work with Tickets (not with Ticket..Dto) - you might get LazyException unless you do some kind of fetching...

**Review by fyarkiy — APPROVED** [review]
Excellent work! Considering Test for employer...

**Review by dmytro-hryhoruk — APPROVED** [review]
Good job

**Pull request merged**